src/modules/functions.js

```diff
@@ -323,6 +323,328 @@ export const displayUserNameOverlay = () => {
+                node.textContent = `$${localCurrencyValue}`; // Replace "₣" node content with the converted value
+                nodeList[index + 1].textContent = ''; // Clear the next node that held the number
+              }
+            }
+          } else if (text.startsWith("₣")) {
+            // For cases where ₣ and the number are already in the same node
+            const tokenValue = parseFloat(text.slice(1));
+            if (!isNaN(tokenValue)) {
+              const localCurrencyValue = (tokenValue * tokenToUsdRate * cfg.usdExchangeRate).toFixed(2);
+              node.textContent = `$${localCurrencyValue}`;
+            }
+          }
+        }
+      });
+    } else {
+      // Standard handling for elements with a single token value
+      const originalText = element.textContent.trim();
+      const tokenMatch = originalText.match(/₣(\d+(\.\d+)?)/); // Match ₣ followed by a number
+
+      if (tokenMatch) {
+        const tokenValue = parseFloat(tokenMatch[1]); // Extract the numeric portion
+        if (!isNaN(tokenValue)) {
+          const localCurrencyValue = (tokenValue * tokenToUsdRate * cfg.usdExchangeRate).toFixed(2);
+          element.innerHTML = originalText.replace(tokenMatch[0], `$${localCurrencyValue}`);
+        }
+      }
+    }
+
+    // Fix width on SFX and Token Modal
+    if (element.classList.contains("tts-modal_tokens__yZ5jv") || element.classList.contains("sfx-modal_tokens__i1DhV")) {
+      element.style.width = "135px";
+    }
+  };
+
+
+  const revertToOriginalTokens = (element) => {
+    const originalContent = element.getAttribute("data-original");
+    if (originalContent) {
+      element.innerHTML = originalContent; // Restore original content
+      element.removeAttribute("data-original"); // Clean up to avoid reprocessing
+    }
+    // Fix width on SFX and Token Modal
+    if (element.classList.contains("tts-modal_tokens__yZ5jv") || element.classList.contains("sfx-modal_tokens__i1DhV")) {
+      element.style.width = "96px";
+    }
+  };
+
+  const processElements = () => {
+    const selectors = [
+      ".top-bar-user_tokens__vAwEj",
+      ".tts-modal_tokens__yZ5jv",
+      ".sfx-modal_tokens__i1DhV",
+      ".get-fishtoys-modal_cost__e3dHa",
+      ".get-tokens-modal_tokens__LX5HO",
+      ".confirm-modal_body__LQQc6 span" // Target modal's span elements for token conversion
+    ];
+
+    selectors.forEach((selector) => {
+      document.querySelectorAll(selector).forEach((element) => {
+        if (element.closest('.get-fishtoys-modal_fishtoy__XFh5h.get-fishtoys-modal_bigtoy__LOwwY')) return;
+        if (toggle) {
+          convertTokensToLocalCurrency(element);
+        } else {
+          revertToOriginalTokens(element);
+        }
+      });
+    });
+  };
+
+  // MutationObserver to process newly added elements, including tokens in dynamically created modals
+  const observer = new MutationObserver((mutations) => {
```

Comment:
please use the existing modal observer in `observers.js`

Reply:
Moved Observer to Observers.js (Cant use existing as it needs to access items in both modals and the main body)